core/trino-main/src/main/java/io/trino/type/DoubleOperators.java

```diff
@@ -175,13 +179,35 @@ public static long castToReal(@SqlType(StandardTypes.DOUBLE) double value)
     @SqlType("varchar(x)")
     public static Slice castToVarchar(@LiteralParameter("x") long x, @SqlType(StandardTypes.DOUBLE) double value)
     {
-        String stringValue = String.valueOf(value);
+        String stringValue;
+
+        // handle positive and negative 0
+        if (value == 0e0) {
+            if (1e0 / value > 0) {
```

Comment:
or, arguably more directly, `Double.compare(0.0, value) == 0`
the `Double.compare` sorts negative zero before positive zero

Reply:
I think it's clearer the way it is. `Double.compare` would require a comment.